### DISCUSSION TITLE
Feature request: text wrapping on paste

### DISCUSSION THREAD

**Discussion by natobsid** [xournalpp/xournalpp#5091] [Ideas]
Hi there,

Firstly, the obligatory thanks-- Xournal++ is a wonderful piece of software that IMHO outperforms almost all of the commercial options available, thank you.

Unfortunately, I have a use case scenario which makes it almost impossible to use. I am a teacher and use note programs as a whiteboard while giving classes, online and in the classroom. I have to be able to import text from other documents easily, and the lack of text wrap when pasting in text from other pages/docs is unfortunately a deal breaker in this context, as I don't have the time to reformat the text. This is a real shame as otherwise Xournal++ would be hands down the best piece of software for the task.

Searching the forums, I see that this has been requested before, though I realize it is probably not necessary for the majority of users. I don't know how easy it would be to implement but is there are chance we might see it in the not too distant future?

Thanks again for all the hard work,

Nathan

**Comment by rolandlo** [comment]
The text editor in Xournal++ is currently pretty primitive. It is worked on currently in #4938 though and it will be much improved. Text wrapping is not included in that PR, but that could be a natural follow-up. Technically it's not so difficult to add text-wrapping, but it will affect the file-format and raises questions with backwards compatibility.
Inserting line breaks on paste in a way that lines don't get too long could also be achieved via a plugin. The text would be read from the clipboard, line breaks would be inserted following some rules. But that is not very flexible if you want to add a word or change the font size after inserting the text.

**Reply by natobsid** [reply]
Thanks for the reply,

[#4938](https://github.com/xournalpp/xournalpp/pull/4938) looks interesting! I have to say I for one would be very happy with either Markdown or the current approach—I don't think it needs to have too many bells and whistles to be effective. 

Re your suggestion of a plugin, that would also be workable. In fact, I have been playing around with something similar this week with my classes in order to use Xournal++. For the most part, inserting line breaks as I type is not an issue, as long text examples are quite rare and the odd break is easy enough. For text pasting and insertion I have started using an [online word wrap tool](https://texttools.org/word-wrap) in a separate browser window (not shared with students). When copying text, I pass it through the tool before copying it into Xournal++. This is a bit tedious but works fine and the students hardly notice the delay. The resulting text contains line breaks and is not real wrapping, but the important thing is just to be able to get a paragraph of text on the page that students can see in the shared window. So a plugin with similar behaviour would absolutely be more than good enough! Unfortunately, my near non-existent programming skills are not up to that task, else I'd be happy to try.

Looking forward to seeing the progress on the text editor. Meanwhile, I'll use my 'hack' for the wrapping. Xournal++ is proving highly usable in every other regard. One thing that's impressed me is how small its processor footprint is. On my old macbook, the lag when using a drawing tablet in other note apps on zoom made stylus work impossible, but with Xournal++ it's as smooth as silk. Being able to annotate PDFs (with layers!) in real time video conferencing is proving really productive. 

Thanks again
